File: src/main/java/org/taktik/connector/business/chapterIV/wrapper/factory/XmlObjectFactory.java
package org.taktik.connector.business.chapterIV.wrapper.factory;

import org.taktik.connector.business.chapterIV.wrapper.Chap4MedicalAdvisorAgreementRequestWrapper;
import org.taktik.connector.business.chapterIV.wrapper.SealedRequestWrapper;
import org.taktik.connector.business.chapterIV.wrapper.UnsealedRequestWrapper;

public interface XmlObjectFactory {
   SealedRequestWrapper createSealedRequest();

   UnsealedRequestWrapper createUnsealedRequest();

   Chap4MedicalAdvisorAgreementRequestWrapper createChap4MedicalAdvisorAgreementRequest();

   String getSubtypeNameToRetrieveCredentialTypeProperties();
}

File: src/main/kotlin/org/taktik/connector/business/chapterIV/wrapper/factory/XmlObjectFactory.kt
package org.taktik.connector.business.chapterIV.wrapper.factory

import org.taktik.connector.business.chapterIV.wrapper.Chap4MedicalAdvisorAgreementRequestWrapper
import org.taktik.connector.business.chapterIV.wrapper.SealedRequestWrapper
import org.taktik.connector.business.chapterIV.wrapper.UnsealedRequestWrapper

interface XmlObjectFactory {
    fun createSealedRequest(): SealedRequestWrapper<*>
    fun createUnsealedRequest(): UnsealedRequestWrapper<*>
    fun createChap4MedicalAdvisorAgreementRequest(): Chap4MedicalAdvisorAgreementRequestWrapper<*>
    fun getSubtypeNameToRetrieveCredentialTypeProperties(): String
}
